code/modules/mob/living/living.dm

```diff
@@ -471,6 +471,10 @@
 
 ///Proc to hook behavior to the change of value in the resting variable.
 /mob/living/proc/set_resting(new_resting, silent = TRUE, instant = FALSE)
+	if(!(mobility_flags & MOBILITY_REST))
+		get_up(instant)
+		update_resting()
```

Comment:
```suggestion
```
This shouldn't be necessary if they never rest.
And if they do, then the cause needs to be found.